### PR TITLE
Misc IDE related configs

### PR DESCRIPTION
**Pull request by shigeya**
I'm getting tired of seeing these. Please merge.
I guessed ES6 is the least language requirement.

- Adding .vscode to .gitignore for ignoring VS Code's configs
  (Duplicate `config.json` removed also.)
- Set ES6 as a baseline in .jsinitrc

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

**Comment by OR13** [comment]
there is now a merge conf.

**Review by msporny — APPROVED** [review]

**Comment by davidlehn** [comment]
The comment and commit say ".jsinitrc", but the edits are to ".jshintrc".  If it's supposed to be .jshintrc, why isn't eslint being used?

**Comment by shigeya** [comment]
> The comment and commit say ".jsinitrc", but the edits are to ".jshintrc". If it's supposed to be .jshintrc, why isn't eslint being used?

Right. Commit message was wrong.. :<

That's due to I'm trying to use up-to-date technology and I'm not up with the latest toolchain yet.
I followed suggestion of VS Code, and I'm using eslint for other my personal project.. I mean, no strong preference.

**Comment by shigeya** [comment]
Added .eslintrc.json also (left .jshintrc also)
Also, the commit message fixed :<

**Comment by shigeya** [comment]
And actually, there are tons of warnings of the use of `async () =>`. Do we want to bump to ES8?

> 'async functions' is only available in ES8 (use 'esversion: 8'). (W119)jshint(W119)

**Comment by OR13** [comment]
yes, probably want to bump to es 8

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "esversion": 6
```
Comment (OR13) [inline .jshintrc:2]:
```suggestion
    "esversion": 8
```

**Comment by shigeya** [comment]
Bumped to ES8

**Pull request merged**